client-spark/common/src/main/java/org/apache/spark/shuffle/writer/WriteBufferManager.java

```diff
@@ -321,7 +349,32 @@ public List<AddBlockEvent> buildBlockEvents(List<ShuffleBlockInfo> shuffleBlockI
 
   @Override
   public long spill(long size, MemoryConsumer trigger) {
-    return 0L;
+    // If it is triggered by other consumers, it will spilled async by current consumer.
+    if (trigger != this) {
+      spillTriggered.set(true);
```

Comment:
Will the variable accessed by multi threads?

Reply:
Yes, It will